applications/CompressiblePotentialFlowApplication/python_scripts/potential_flow_solver.py

```diff
@@ -169,67 +160,84 @@ def AddVariables(self):
-                self.settings["reform_dofs_at_each_step"].GetBool(),
-                self.settings["calculate_solution_norm"].GetBool(),
-                self.settings["move_mesh_flag"].GetBool())
-        elif strategy == "non_linear":
-            conv_criteria = KratosMultiphysics.ResidualCriteria(
-                self.settings["relative_tolerance"].GetDouble(),
-                self.settings["absolute_tolerance"].GetDouble())
-            max_iterations = self.settings["maximum_iterations"].GetInt()
-
-            self.solver = KratosMultiphysics.ResidualBasedNewtonRaphsonStrategy(
-                self.GetComputingModelPart(),
-                time_scheme,
-                self.linear_solver,
-                conv_criteria,
-                max_iterations,
-                self.settings["compute_reactions"].GetBool(),
-                self.settings["reform_dofs_at_each_step"].GetBool(),
-                self.settings["move_mesh_flag"].GetBool())
-        else:
-            raise Exception("Element not implemented")
+        self.ComputeNodalElementalNeighbours()
 
-        (self.solver).SetEchoLevel(self.settings["echo_level"].GetInt())
-        self.solver.Initialize()
+        solution_strategy = self._GetSolutionStrategy()
+        solution_strategy.SetEchoLevel(self.settings["echo_level"].GetInt())
+        solution_strategy.Initialize()
+
+        KratosMultiphysics.Logger.PrintInfo(self.__class__.__name__, "Solver initialization finished.")
 
     def AdvanceInTime(self, current_time):
         raise Exception("AdvanceInTime is not implemented. Potential Flow simulations are steady state.")
 
-    def _ComputeNodalNeighbours(self):
+    def ComputeNodalElementalNeighbours(self):
         # Find nodal neigbours util call
-        KratosMultiphysics.FindNodalNeighboursProcess(self.main_model_part).Execute()
+        data_communicator  = KratosMultiphysics.DataCommunicator.GetDefault()
+        KratosMultiphysics.FindGlobalNodalElementalNeighboursProcess(
+            data_communicator,
+            self.GetComputingModelPart()).Execute()
 
     def _GetStrategyType(self):
         element_type = self.settings["formulation"]["element_type"].GetString()
         if "incompressible" in element_type:
             if not self.settings["formulation"].Has("penalty_coefficient"):
-                strategy = "linear"
+                strategy_type = "linear"
             elif self.settings["formulation"]["penalty_coefficient"].GetDouble() == 0.0:
-                strategy = "linear"
+                strategy_type = "linear"
             else:
-                strategy = "non_linear"
+                strategy_type = "non_linear"
         elif "compressible" in element_type:
-            strategy = "non_linear"
+            strategy_type = "non_linear"
         else:
-            strategy = ""
+            strategy_type = None
+        return strategy_type
 
-        return strategy
+    @classmethod
```

Comment:
```suggestion

```

I think adding this is wrong, but I am not entirely sure
I mean, in the baseclass this is not a `classmethod`

Reply:
The point is that this method could be a function, as it does not require the `self`. This turns into a Python warning.
If I'm not wrong the purpose of the `@classmethod` decorator is to explicitly indicate that the function is a method.

Reply:
> This turns into a Python warning.

well we ignore this often enough ;)

if you are sure this works/is ok then please go ahead